client/client-multi/client-cpp/impl/ClientStatus.cpp

```diff
@@ -420,51 +420,59 @@ SequenceNumber ClientStatus::getAppSeqNumber() const
     return appSeqNumberDefault_;
 }
 
-void ClientStatus::setAppSeqNumber(SequenceNumber appSeqNumber)
+template< ClientParameterT Type, class ParameterData >
+void ClientStatus::setParameterData(const ParameterData& data)
 {
-    auto parameter_it = parameters_.find(ClientParameterT::APPSEQUENCENUMBER);
+    auto parameter_it = parameters_.find(Type);
     if (parameter_it != parameters_.end()) {
-        parameter_it->second->setValue(appSeqNumber);
+        parameter_it->second->setValue(data);
+        hasUpdate_ = true;
     }
 }
 
-bool ClientStatus::isRegistered() const
+template< ClientParameterT Type, class ParameterData >
+ParameterData ClientStatus::getParameterData(const ParameterData& defaultValue) const
 {
-    auto parameter_it = parameters_.find(ClientParameterT::ISREGISTERED);
+    auto parameter_it = parameters_.find(Type);
     if (parameter_it != parameters_.end()) {
-        return boost::any_cast<bool>(parameter_it->second->getValue());
+        return boost::any_cast<ParameterData>(parameter_it->second->getValue());
     }
-    return isRegisteredDefault_;
+
+    return defaultValue;
+}
+
+void ClientStatus::setAppSeqNumber(SequenceNumber appSeqNumber)
+{
+    setParameterData<ClientParameterT::APPSEQUENCENUMBER>(appSeqNumber);
+}
+
+bool ClientStatus::isRegistered() const
+{
+    return getParameterData<ClientParameterT::ISREGISTERED>(isRegisteredDefault_);
 }
 
 void ClientStatus::setRegistered(bool isRegisteredP)
 {
     if (isRegistered() !=  isRegisteredP) {
-        auto parameter_it = parameters_.find(ClientParameterT::ISREGISTERED);
-        if (parameter_it != parameters_.end()) {
-            parameter_it->second->setValue(isRegisteredP);
-        }
+        setParameterData<ClientParameterT::ISREGISTERED>(isRegisteredP);
     }
 }
 
 std::string ClientStatus::getEndpointAccessToken()
 {
-    auto parameter_it = parameters_.find(ClientParameterT::EP_ACCESS_TOKEN);
-    if (parameter_it != parameters_.end()) {
-        auto token = boost::any_cast<std::string>(parameter_it->second->getValue());
-        if (!token.empty()) {
-            return token;
-        }
+    std::string token = "";
```

Comment:
There is no need to initialize the variable explicitly. For this purposes std::string has default constructor.